src/coreclr/src/tools/Common/Compiler/DependencyAnalysis/SortableDependencyNode.cs

```diff
@@ -31,6 +31,9 @@ public abstract class SortableDependencyNode : DependencyNodeCore<NodeFactory>,
         /// </remarks>
         public abstract int ClassCode { get; }
 
+        // Custom sort order. Used to override the default sorting mechanics.
+        public int CustomSort = int.MaxValue;
```

Comment:
Could you move SortableDependencyNode out of Common and into the ReadyToRun specific portion of the compiler?

Reply:
I think it should be a property not a field

Reply:
@MichalStrehovsky Could you provide a rationale as to why we should move this from Common? I believe that the CoreRT compiler still uses this exact mechanism, and providing a means to adjust the sort order is likely useful for the CoreRT system as well.

Reply:
We don't want externally mutable state on Dependency nodes and this adds mutable state to all of them - I don't want anyone to use that as a precedent. Broken windows theory and all.

CoreRT doesn't need this because the linker is going to shuffle things anyway.

Reply:
(If I want to sort something in the middle of the compilation, I better remember that sort order is not deterministic until whatever sets this does it's job. I don't think it's a good property. The sorting should be external to DependencyNodes but I don't want to ask you to do more work, so it's easier to just unshare)

Reply:
Ok, fair.

Reply:
I've made the SortableDependencyNode into a partial class. Sortable dependency node is tied to quite a lot of the rest of the common code, so it isn't practical to extract.